src/Patches/QuestPatch.cs

```diff
@@ -42,6 +42,7 @@ private static void After_getQuestFromId(int id, ref Quest __result)
                     {
                         // This is fix for use custom dialogue text for slay monster quest type
                         (__result as SlayMonsterQuest).dialogueparts.Clear();
+                        (__result as SlayMonsterQuest).reward.Value = customQuest.Reward;
```

Comment:
👍

Reply:
This is my junk (originally I written it with the `as` operator), but in this case it can be refactored in this way:

```cs
if (customQuest.BaseType == QuestType.Monster && __result is SlayMonsterQuest slayQuest)
{
   // This is fix for use custom dialogue text for slay monster quest type
   slayQuest.dialogueparts.Clear();
   slayQuest.reward.Value = customQuest.Reward;
}
```

This looks better and I wanted refactor it in this way, but I didn't because I left modding and focused on my game project. If you want, you can refactor it.

Reply:
Changed it 👍